web/src/main/java/org/cbioportal/proxy/ProxyController.java

```diff
@@ -1,38 +1,74 @@
 package org.cbioportal.proxy;
 
-import org.springframework.http.HttpEntity;
-import org.springframework.http.HttpHeaders;
-import org.springframework.http.HttpMethod;
-import org.springframework.http.MediaType;
+import org.cbioportal.web.util.ResettableHttpServletRequestFilter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.*;
 import org.springframework.http.converter.StringHttpMessageConverter;
+import org.springframework.util.StringUtils;
 import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
 import org.springframework.web.client.RestTemplate;
 
+import javax.annotation.PostConstruct;
 import javax.servlet.http.HttpServletRequest;
 import java.net.URI;
 import java.net.URISyntaxException;
 import java.nio.charset.StandardCharsets;
 
 @RestController
 public class ProxyController {
+    @Value("${show.oncokb:true}")
+    private Boolean showOncokb;
+    @Value("${oncokb.token:}")
+    private String oncokbToken;
+    @Value("${oncokb.public_api.url:}")
+    private String oncokbPublicApiUrl;
+
+    private Logger LOG = LoggerFactory.getLogger(ProxyController.class);
 
     @RequestMapping("/**")
     public String proxy(@RequestBody(required = false) String body, HttpMethod method, HttpServletRequest request)
         throws URISyntaxException {
+        HttpHeaders httpHeaders = new HttpHeaders();
+        String contentType = request.getHeader("Content-Type");
+        if (contentType != null) {
+            httpHeaders.setContentType(MediaType.valueOf(contentType));
+        }
+        return exchangeData(body, buildUri(request.getPathInfo(), request.getQueryString(), false), method, httpHeaders).getBody();
+    }
 
-        String queryString = request.getQueryString();
-        URI uri = new URI("http:/" + request.getPathInfo() + (queryString == null ? "" : "?" + queryString));
+    @RequestMapping("/oncokb/**")
+    public ResponseEntity<String> proxyOncokb(@RequestBody(required = false) String body, HttpMethod method, HttpServletRequest request)
+        throws URISyntaxException {
+        if (!this.showOncokb) {
+            return new ResponseEntity<>("OncoKB service is disabled.", HttpStatus.NOT_FOUND);
+        }
 
         HttpHeaders httpHeaders = new HttpHeaders();
         String contentType = request.getHeader("Content-Type");
         if (contentType != null) {
             httpHeaders.setContentType(MediaType.valueOf(contentType));
         }
+        if (!StringUtils.isEmpty(this.oncokbToken)) {
+            httpHeaders.add("Authorization", "Bearer " + this.oncokbToken);
+        }
+//        else if(StringUtils.isEmpty(this.oncokbToken) && this.oncokbPublicApiUrl.equalsIgnoreCase("legacy.oncokb.org")){
```

Comment:
@sheridancbio I commented out this section since I don't want to log a warning everytime the proxy is used. What's the best way to print a warning message after instance is deployed?

Reply:
Code can be added that will run at application startup using the interface "ServletContextListener"
https://javaee.github.io/javaee-spec/javadocs/javax/servlet/ServletContextListener.html
https://stackoverflow.com/questions/4366074/java-web-application-initialization-and-shutdown
you need to add a reference to the implementing class in web.xml